### PR TITLE
Feature/fix tool date

### PR DESCRIPTION
**Pull request by garyluu**
Previously https://github.com/dockstore/dockstore/pull/3374 didn't test for hosted tool deletion of a default version. This adds the test and fixes it.  This allows ~my PR and~ Andrew's PR on the UI to pass (since it requires a new webservice).  Everyone else is unaffected for now.

~Push cancelled on purpose.~ Restarting push since Travis isn't so busy now.

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tag.java [hunk 1]
```diff
@@ -90,7 +91,7 @@ public Version createEmptyVersion() {
 
     @Override
     public Date getDate() {
-        return this.getLastBuilt();
+        return Stream.of(this.getLastBuilt(), this.getDbCreateDate()).filter(Objects::nonNull).findFirst().orElse(null);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tag.java:94]: 👍 Hadn't seen that technique before.

Comment (garyluu) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tag.java:94]: First thing I could think of, could've sworn there was a easier way. 🤷

Comment (agduncan94) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tag.java:94]: Were you thinking of the Apache method? http://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/ObjectUtils.html#firstNonNull%28T...%29

Comment (denis-yuen) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tag.java:94]: firstNonNull or similar does seem more readable

**Pull request merged**